### PR TITLE
Unit tests updownload

### PR DESCRIPTION
**Pull request by chStaiger**
- Added some local test data
- Added a test to check the generation of iRODS paths when a folder is uploaded

The gneration of local download paths when a collection is downloaded needs to go into the integration tests, since it is dependent on an iRODS database query.

### CODE REVIEW

**Comment by chStaiger** [comment]
Now the path wrangling for uploads from windows to irods also pass the unit tests. I found the bug. Good that we have unit tests 😄

File: ibridges/irodsconnector/data_operations.py [hunk 1]
```diff
@@ -185,25 +199,30 @@ def _upload_collection(session: Session, local_path: Union[str, Path],
...
-    _ = create_collection(session, upload_path)
-
-    for folder, file_name in paths:
-        # ensure iRODS collection exists
-        _ = create_collection(session, upload_path.joinpath(folder.lstrip('/')))
-        # call irods put for Path(local_path, f[0], f[1]) to destination collection
-        dest = IrodsPath(session, str(upload_path), folder.lstrip('/'))
+    source_to_dest = _create_irods_dest(local_path, irods_path)
+    for source, dest in source_to_dest:
+        _ = create_collection(session, dest.parent)
```
Comment (qubixes) [inline ibridges/irodsconnector/data_operations.py:204]: One thing that we might have to worry about is absolute/relative paths. I have tested a bit with pathlib, and something like `Path(".").parent` gives the same directory as `Path(".")`. I currently can't oversee if is an issue here specifically, but might be good to keep in mind.

**Review by qubixes — APPROVED** [review]
Looks very reasonable. I'm assuming most of the data_ops functions are going to be tested in the integration tests?

**Comment by chStaiger** [comment]
> Looks very reasonable. I'm assuming most of the data_ops functions are going to be tested in the integration tests?

Indeed. In the integration tests we need the `_obj_put` and `_obj_get` and for the download of the collections also the path wrangling test for `_create_local_dest`.
With those tests all up and downloads should work fine.

**Pull request merged**